azure: add an option to allow enabling/disabling secondary ip configuration (#4432)
canonical/cloud-init: doc/rtd/reference/datasources/azure.rst

  Boolean set to True to use network configuration described by Azure's IMDS
  endpoint instead of fallback network config of DHCP on eth0. Default is
  True.
* :command:`apply_network_config_for_secondary_ips`

>   Boolean set to True to configure primary and secondary IP addresses for each

Reviewer: ```suggestion
  Configure secondary IP addresses for each NIC.
  Default is True.
```

Reviewer: Is there a standard template for documenting these options?

Reviewer 2: Unfortunately no. We generally try to keep things consistent, but there's no template or specific rules for it. I'll see if that's something we can add.

Author: done!
